crates/circuit/src/packed_instruction.rs

```diff
@@ -628,136 +674,79 @@ impl PackedInstruction {
             op: gate.into(),
             qubits,
             clbits: Default::default(),
-            params,
+            params: params.map(|params| Box::new(Parameters::Params(*params))),
             label: None,
             #[cfg(feature = "cache_pygates")]
             py_op: OnceLock::new(),
         }
     }
 
-    /// Access the standard gate in this `PackedInstruction`, if it is one.  If the instruction
-    /// refers to a Python-space object, `None` is returned.
-    #[inline]
-    pub fn standard_gate(&self) -> Option<StandardGate> {
-        self.op.try_standard_gate()
+    /// Pack a [ControlFlow] operation with blocks into a complete instruction.
+    pub fn from_control_flow(
+        control_flow: ControlFlow,
+        blocks: Vec<Block>,
+        qubits: Interned<[Qubit]>,
+        clbits: Interned<[Clbit]>,
+        label: Option<String>,
+    ) -> Self {
+        Self {
+            op: control_flow.into(),
+            qubits,
+            clbits,
+            params: Some(Box::new(Parameters::Blocks(blocks))),
+            label: label.map(Box::new),
+            #[cfg(feature = "cache_pygates")]
+            py_op: Default::default(),
+        }
     }
 
     /// Get a slice view onto the contained parameters.
     #[inline]
     pub fn params_view(&self) -> &[Param] {
```

Comment:
I'm not _100%_ convinced that this should be `&[Param]` and not `Option<&[Param]>`, but I can certainly believe it minimises the API changes, so let's stick with it - it's also not at all clear to me that `Option<&[Param]>` would cause any benefit other than noise.